# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/0ad/hotfix-t2.patch
# Copyright (C) 2022 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- /dev/null
+++ 0ad-0.27.0/libraries/source/spidermonkey/patches/t2.patch	2025-04-26 15:53:34.885000000 +0200
@@ -0,0 +1,30 @@
+--- firefox-103.0/build/moz.configure/rust.configure.vanilla	2022-08-09 12:08:08.707934940 +0200
++++ firefox-103.0/build/moz.configure/rust.configure	2022-08-09 12:18:52.980909250 +0200
+@@ -394,7 +394,7 @@
+         if len(narrowed) == 1:
+             return narrowed[0].rust_target
+ 
+-        return None
++        return candidates[1].rust_target
+ 
+     rustc_target = find_candidate(candidates)
+ 
+--- mozjs-115.16.1/build/autoconf/config.guess.vanilla	2025-04-26 15:49:54.179082764 +0200
++++ mozjs-115.16.1/build/autoconf/config.guess	2025-04-26 15:50:11.556268838 +0200
+@@ -1028,7 +1028,7 @@
+ 	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+ 	;;
+     i*86:Linux:*:*)
+-	GUESS=$UNAME_MACHINE-pc-linux-$LIBC
++	GUESS=$UNAME_MACHINE-t2-linux-$LIBC
+ 	;;
+     ia64:Linux:*:*)
+ 	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
+@@ -1160,7 +1160,7 @@
+ 		LIBCABI=${LIBC}x32
+ 	    fi
+ 	fi
+-	GUESS=$UNAME_MACHINE-pc-linux-$LIBCABI
++	GUESS=$UNAME_MACHINE-t2-linux-$LIBCABI
+ 	;;
+     xtensa*:Linux:*:*)
+ 	GUESS=$UNAME_MACHINE-unknown-linux-$LIBC
--- 0ad-0.27.0/libraries/source/spidermonkey/patches/patch.sh.vanilla	2025-04-26 15:43:15.665994609 +0200
+++ 0ad-0.27.0/libraries/source/spidermonkey/patches/patch.sh	2025-04-26 15:43:49.250445149 +0200
@@ -17,6 +17,9 @@
 # Add needed debug define in pkg-config file.
 patch -p1 <"${PATCHES}"/FixPkgConfigDebug.diff
 
+# Fix t2 rustc target
+patch -p1 <"${PATCHES}"/t2.patch
+
 # On macOS, embedded build is broken due to a faulty check for pkg-config
 # https://bugzilla.mozilla.org/show_bug.cgi?id=1776255
 # The fix above is included in ESR 128, but does not apply to 115
